### PR TITLE
[NUI] Make EllipsisPosition property public

### PR DESCRIPTION
**Pull request by shrouqsabah**
Signed-off-by: Shrouq Sabah s.sabah@samsung.com
### Description of Change ###

Public open API list

- NUIConstants
    -  EllipsisPosition {End, Start, Middle}   // Enum

- TextLabel
    -  EllipsisPosition { get; set; } // Property

- TextField
    -  EllipsisPosition { get; set; }      // Property

- TextEditor
    -  EllipsisPosition { get; set; }    // Property
    -  Ellipsis { get; set; }   

  
### API Changes ###
 
- ACR: TCSACR-444
- https://code.sec.samsung.net/jira/browse/TCSACR-444

### CODE REVIEW

File: src/Tizen.NUI/src/public/Common/NUIConstants.cs [hunk 1]
```diff
@@ -887,7 +887,7 @@ public enum VerticalLineAlignment
     /// <summary>
     /// An enum of ellipsis position.
     /// </summary>
-    [EditorBrowsable(EditorBrowsableState.Never)]
+    /// <since_tizen> 9 </since_tizen>
     public enum EllipsisPosition
     {
         /// <summary>
```
Comment (wonrst) [inline src/Tizen.NUI/src/public/Common/NUIConstants.cs:893]: End, Start, and Middle enums should also be opened.

File: src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs [hunk 2]
```diff
@@ -1419,7 +1419,7 @@ public bool Ellipsis
         /// The ellipsis position type when the text size over the layout size.<br />
```
Comment (wonrst) [inline src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs:1419]:
```
TextEditor
EllipsisPosition { get; set; } // Property
Ellipsis { get; set; }
```

You also need to open the TextEditor's Ellipsis Property.

File: src/Tizen.NUI/src/public/Common/NUIConstants.cs (hunk 1 above)
Comment (shrouqsabah) [reply src/Tizen.NUI/src/public/Common/NUIConstants.cs:893]: @wonrst , done

File: src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs (hunk 2 above)
Comment (shrouqsabah) [reply src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs:1419]: @wonrst , done

File: src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs [hunk 3]
```diff
@@ -1419,7 +1419,7 @@ public bool Ellipsis
...
         /// The ellipsis position: End, Start or Middle.<br />
```
Comment (wiertel) [inline src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs:1420]: The same description is used in other places
```suggestion
        /// Specifies which portion of the text should be replaced with an ellipsis when the text size exceeds the layout size.<br />
        /// The ellipsis position: End, Start or Middle.
```
The last sentence could also be removed, for a person reading the docs it's only 1 click to see details of the type.

File: src/Tizen.NUI/src/public/Common/NUIConstants.cs [hunk 4]
```diff
@@ -887,25 +887,25 @@ public enum VerticalLineAlignment
...
 
         /// <summary>
         /// ellipsis position at start.
         /// </summary>
-        [EditorBrowsable(EditorBrowsableState.Never)]
+        /// <since_tizen> 9 </since_tizen>
         Start,
 
         /// <summary>
         /// ellipsis position in middle.
```
Comment (wiertel) [inline src/Tizen.NUI/src/public/Common/NUIConstants.cs:906]: in the middle

File: src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs (hunk 3 above)
Comment (shrouqsabah) [reply src/Tizen.NUI/src/public/BaseComponents/TextEditor.cs]: Done

File: src/Tizen.NUI/src/public/Common/NUIConstants.cs (hunk 4 above)
Comment (shrouqsabah) [reply src/Tizen.NUI/src/public/Common/NUIConstants.cs]: Done

**Comment by dongsug-song** [comment]
ACR has been accepted, so merge now

**Review by dongsug-song — APPROVED** [review]
ACR has been accepted

**Pull request merged**